pkg/apis/storage/types.go

```diff
@@ -293,8 +293,34 @@ type CSIDriverSpec struct {
 	// "csi.storage.k8s.io/pod.uid": string(pod.UID)
 	// +optional
 	PodInfoOnMount *bool
+
+	// If set, Mode defines how this CSI volume driver can be used. The default
+	// is "persistent", which is the usage defined by the CSI specification.
+	// Other modes are:
+	// - "ephemeral": only use this driver for inline ephemeral volumes (CSIVolumeSource)
+	// - "persistent+ephemeral": both modes supported
+	// More modes may be added in the future.
+	// +optional
+	Mode *CSIDriverMode
```

Comment:
Validation for this new field needs to be added in https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/storage/validation/validation.go

Reply:
Agreed, will add this. I just wanted to fix the roundtrip error first before working on making the API nicer to use.

Reply:
Done.